server/security/services/keycloak/schemaDirectives/hasRole.js

```diff
@@ -22,16 +22,21 @@ class HasRoleDirective extends SchemaDirectiveVisitor {
         throw newInternalServerError(context)
       }
 
-      const token = context.auth.getToken()
+      if (!context.auth || !context.auth.isAuthenticated()) {
+        const AuthorizationErrorMessage = `Unable to find authentication. Authorization is required for field ${field.name} on parent ${info.parentType.name}. Must have one of the following roles: [${roles}]`
```

Comment:
Right now, I don't see how this condition could occur, because if Keycloak is enabled, the GraphQL endpoint in the server is protected using a keycloak express middleware. 

This means that when keycloak is enabled, the server cannot be accessed unless the user is authenticated, so this code path **shouldn't be** reachable. 

That being said, it may enable a use case later on where it's possible to do certain things while authenticated/not authenticated. So +1 on this.

Reply:
> That being said, it may enable a use case later on where it's possible to do certain things while authenticated/not authenticated. So +1 on this.

yes, that was my thinking too.